tests/compatibility/ConnectionRealTestCase.nut

```diff
@@ -0,0 +1,75 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+@include __PATH__+"/../Base.nut"
+
+// ConnectionRealTestCase
+// Tests for MessageManager constructor options (onPartnerConnected, onConnectedReply, connectionManager)
+// Test connection with real ConnectionManager
+class ConnectionRealTestCase extends ImpTestCase {
+
+    function setUp() {
+        infoAboutSide();
+    }
+    
+    function testConnection() {
+        if (isAgentSide()) {
```

Comment:
Two questions here:
1. Why do we need this test for Agent if it doesn't do anything useful (but is even harmful - see 2)?
2. In the case when the code runs on the agent side, mm is not defined which results in `[+616.59/0.97s warning] Agent Runtime Error: ERROR: no handler for agent.send("MM_CONNECT", data)` error message. This is definitely a wrong test behavior.